rotkehlchen/api/rest.py

```diff
@@ -1984,3 +2003,23 @@ def sync_data(
             status_code = HTTPStatus.OK
 
         return api_response(result_dict, status_code=status_code)
+
+    @require_loggedin_user()
+    def import_data_from_file(
+            self,
+            source: Literal['cointracking.info', 'crypto.com'],
+            file: FileStorage,
+    ) -> Response:
+        if file.filename is None or not file.filename.endswith('.csv'):
+            result = wrap_in_fail_result(f'{file.filename} is not a csv')
+            return api_response(result, status_code=HTTPStatus.BAD_REQUEST)
+
+        temp_directory = Path(mkdtemp())
```

Comment:
Don't use mkdtemp here. Use https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

We already use it in coinbasepro if you wanna look at an example: https://github.com/rotki/rotki/blob/31cd9b0d5cbe56777a36adafc817fba54abd3d58/rotkehlchen/exchanges/coinbasepro.py#L558

But the python docs are also good.

If you do that the cleanup at the end will not be needed at all.